cmd/cody-gateway/qa/e2e_test.go

```diff
@@ -0,0 +1,159 @@
+package qa
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/sourcegraph/sourcegraph/internal/codygateway"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+func Test_Completions(t *testing.T) {
+	gatewayURL, gatewayToken := parseBackendData(t)
+	for _, f := range []codygateway.Feature{codygateway.FeatureCodeCompletions, codygateway.FeatureChatCompletions} {
+		for name, p := range map[string]GatewayFeatureClient{"anthropic": AnthropicGatewayFeatureClient{}, "fireworks": FireworksGatewayFeatureClient{}, "openai": OpenAIGatewayFeatureClient{}} {
+			for _, stream := range []bool{false, true} {
+				t.Run(string(f)+" "+name+" stream "+strconv.FormatBool(stream), func(t *testing.T) {
+					stream := stream
+					// avoid mutating the same URL
+					u := *gatewayURL
+					t.Parallel()
+					req := &http.Request{URL: &u, Header: make(http.Header)}
+					req.Header.Set("X-Sourcegraph-Feature", string(f))
+					req.Header.Set("Authorization", "Bearer "+gatewayToken)
+					req, err := p.GetRequest(f, req, stream)
+					if err != nil && errors.Is(err, errNotImplemented) {
```

Comment:
```suggestion
					if errors.Is(err, errNotImplemented) {
```
non-blocking: I would presume this would work, have you observed it not working like this?

Reply:
Nope, this is probably me adding another condition to an if & not removing the redundant nil check.